### PR TITLE
Simplify module loading s'more.

### PR DESCRIPTION
**Pull request by applmak**
This is a pretty involved change that streamlines the way we refer to loaded, but not instantiated modules.

### CODE REVIEW

File: client/modules/module_manager.js [hunk 1]
```diff
@@ -112,7 +112,7 @@ define(function(require) {
     // Server has asked us to load a new module.
     network.on('loadModule', function(bits) {
       var def = bits.module;
-      var code = bits.def;
+      var code = bits.module.def;
```
Comment (bmt) [inline client/modules/module_manager.js]: Can we do anything about the variable names here so the server / client match up a little better?

Comment (applmak) [inline client/modules/module_manager.js]: I'll add a TODO.

File: server/modules/module_def.js [hunk 2]
```diff
@@ -129,75 +130,109 @@ let loadModuleAtPath = function(path) {
...
+ */
+class ModuleDef {
+  constructor(name, path, title, author, config) {
+    this.name = name;
+    this.path = path;
+    this.config = config || {};
+    this.title = title;
+    this.author = author;
+    
+    // The promise<ModuleDef> that when set, indicates the module is loaded.
```
Comment (bmt) [inline server/modules/module_def.js]: Isn't this promise<string>?

File: server/modules/module_def.js [hunk 3]
```diff
@@ -129,75 +130,109 @@ let loadModuleAtPath = function(path) {
...
-    // This map is only populated during the loading process.
-    this.moduleLoader = {};
-
-    // map of path -> string.
-    this.modules = {};
+/**
+ * The ModuleDef class contains all the information necessary to load & 
+ * instantiate a module, including code location and config parameters.
+ */
+class ModuleDef {
```
Comment (bmt) [inline server/modules/module_def.js]: You should make sure that this and the client equivalent have the same name. I can't remember what I called the class when I refactored the client.

File: server/modules/module_def.js (hunk 2 above)
Comment (applmak) [inline server/modules/module_def.js]: It's actually promise<undefined>...

File: server/modules/module_def.js (hunk 3 above)
Comment (applmak) [inline server/modules/module_def.js]: Added TODO.

File: server/modules/module_def.js (hunk 2 above)
Comment (bmt) [inline server/modules/module_def.js]: Escape your html entities :)

**Pull request merged**